libbeat/autodiscover/builder/helper.go

```diff
@@ -155,41 +156,21 @@ func IsNoOp(hints common.MapStr, key string) bool {
 }
 
 // GenerateHints parses annotations based on a prefix and sets up hints that can be picked up by individual Beats.
-func GenerateHints(annotations common.MapStr, container, prefix string) common.MapStr {
+func GenerateHints(annotations common.MapStr, container, prefix string, separator string) common.MapStr {
+	validPrefix := regexp.MustCompile("^(?P<Module>[a-zA-Z0-9]*)\\.?(" + container + ")?" + separator + "(?P<Key>.*)$")
```

Comment:
The dot before the container, and the container itself should only be expected if a non-empty container name is passed. Maybe we could use different expressions if a container name is passed or not. It shouldn't be needed to pass container names in docker autodiscover as there are no pods there, so annotations always refer to the container itself.

Reply:
True, I didn't consider this. Then I would propose to simply relax hint generation for docker so that it considers hint candidates with slash like before plus all keys that contain only letters and have a mapStr as value. Namespace validation happens already by each beat when these try to pick their namespace from all hints. I think this should even work without an additional configuration option, just some added documentation at the docker level for metric- and filebeat. However it would allow to mix both syntaxes, which should be ok and maybe helpful for migrating back to the original syntax in case AWS relaxes their validation at some point. What do you think?